mlir/lib/Bindings/Python/ExecutionEngineModule.cpp

```diff
@@ -124,6 +124,17 @@ NB_MODULE(_mlirExecutionEngine, m) {
           },
           nb::arg("name"), nb::arg("callback"),
           "Register `callback` as the runtime symbol `name`.")
+      .def(
+          "initialize",
+          [](PyExecutionEngine &executionEngine) {
+            mlirExecutionEngineInitialize(executionEngine.get());
+          },
+          "Initialize the ExecutionEngine. Global constructors specified by "
+          "`llvm.mlir.global_ctors` will be run. One common scenario is that "
+          "kernel binary compiled from `gpu.module` gets loaded during "
+          "initialization. Make sure all symbols are resolvable before "
+          "initialization by calling `raw_register_runtime` or including "
+          "shared libraries.")
```

Comment:
By the way: now that invoke is doing initialize() lazily, is this still useful to expose to python users?

Reply:
> By the way: now that invoke is doing initialize() lazily, is this still useful to expose to python users?

yes it is because python API has `raw_register_runtime` which calls `mlirExecutionEngineRegisterSymbol`. As long as user of one binding can use `mlirExecutionEngineRegisterSymbol` to register symbol, the exposure would be useful. The symbol resolution would fail even the global ctors don't use the unregistered ones(also possible to fix it from this direction if ORC allows us only resolve symbols actually needed by the ctors).

Reply:
> yes it is because python API has raw_register_runtime which calls mlirExecutionEngineRegisterSymbol. As long as user of one binding can use mlirExecutionEngineRegisterSymbol to register symbol, the exposure would be useful. 

I don't follow. Can you write a python test that show the usefulness of the API?

Reply:
Thanks!